metaissue_issuetypes.go

```diff
@@ -0,0 +1,38 @@
+package jira
+
+import (
+	"context"
+	"errors"
+	"fmt"
+)
+
+type GetIssueTypesResponse struct {
+	Values []*MetaIssueType `json:"values,omitempty"`
+	Total  int              `json:"total,omitempty"`
+}
+
+// GetIssueTypesForProject returns the issue types for a given project, and is designed to replace the
+// /rest/api/2/issue/createmeta endpoint, which is called using GetCreateMetaWithOptionsWithContext. That endpoint
+// is deprecated as of Jira Server 9.0.0.
+// See: https://docs.atlassian.com/software/jira/docs/api/REST/9.0.0/#issue-getCreateIssueMetaProjectIssueTypes
+func (s *IssueService) GetIssueTypesForProject(ctx context.Context, projectKey string) (*GetIssueTypesResponse, *Response, error) {
+	if projectKey == "" {
+		return nil, nil, errors.New("project key cannot be empty")
+	}
+
+	apiEndpoint := fmt.Sprintf("rest/api/2/issue/createmeta/%s/issuetypes", projectKey)
```

Comment:
Maybe already on your radar, but you may also want to implement `/rest/api/2/issue/createmeta/{projectIdOrKey}/issuetypes/{issueTypeId} ` - I wasn't sure if we needed details from its response or if just `issueTypes` was enough, but just a heads up!

Reply:
Ah yeah, good call. I'm going to do baby steps, given that releasing this doesn't seem to be _too_ painful.